swim/ping_handler.go

```diff
@@ -39,12 +39,12 @@ func handlePing(node *Node, req *ping) (*ping, error) {
 
 	if req.App == "" {
 		if node.requiresAppInPing {
-			node.logger.Warnf("Rejected ping from unknown ringpop app %s", req.App)
+			node.logger.WithFields("req", req).Warnf("Rejected ping from unknown ringpop app")
```

Comment:
Don't add the full req. It could be huge (due to `Changes`). `Source` might be enough.